### PR TITLE
Adjust PUT method behavior to POST one for default content type in WebCmdlets

### LINKED ISSUE #16318: Invoke-RestMethod doesn't include automatic ContentType value when not using POST method
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

When using a POST, ContentType is automatically set to `application/x-www-form-urlencoded`
```
 invoke-restmethod https://webhook.site/6a5097af-3644-4462-9d92-d910088d2f4d -Method POST -Body @{ Test = "test" }
```

When using other methods, it is not. It is blank. 

```
 invoke-restmethod https://webhook.site/6a5097af-3644-4462-9d92-d910088d2f4d -Method PUT -Body @{ Test = "test" }
```

image

### Expected behavior

```console
I would expect that the content type would default to form encoded for all request methods (or at least for PUT). 
```

### Actual behavior

```console
It is only set to form encoded for POST.
```

### Error details

```console
No errors.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.0-rc.1
PSEdition                      Core
GitCommitId                    7.2.0-rc.1
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
```


### Visuals

_No response_

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
If -ContentType is null, -Headers don't contain `Content-Type` and -Method is POST or PUT --> `WebSession.ContentHeaders[HttpKnownHeaderNames.ContentType] = "application/x-www-form-urlencoded"`

@iSazonov

## PR Context
Fixes #16318

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9863
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
@CarloToso Please add new test like `    It "Validate Invoke-WebRequest default ContentType for CustomMethod POST" {
`

**Comment by iSazonov** [comment]
> I think I found a BUG (line 1491) it should check for `WebSession.ContentHeaders[HttpKnownHeaderNames.ContentType]` instrad of `ContentType` (this should be ok because if both are present ContentType overwrites `WebSession.ContentHeaders[HttpKnownHeaderNames.ContentType]`) unfortunately I don't know how to test it, if confirmed I'll open another PR to fix it

Please open new issue to dicuss.

**Comment by CarloToso** [comment]
Unrelated test failure

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 1]
```diff
@@ -2517,6 +2526,14 @@ Describe "Invoke-RestMethod tests" -Tags "Feature", "RequireAdminOnWindows" {
         $result.Output.Headers.'Content-Type' | Should -Be "application/x-www-form-urlencoded"
     }
 
+    It "Validate Invoke-RestMethod default ContentType for CustomMethod PUT" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:2529]: It is the same code as previous so we can use TestCases.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 2]
```diff
@@ -752,9 +752,14 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
         ($result.Output.Content | ConvertFrom-Json).method | Should -Be "TEST"
     }
 
-    It "Validate Invoke-WebRequest default ContentType for CustomMethod POST" {
-        $uri = Get-WebListenerUrl -Test 'Post'
-        $command = "Invoke-WebRequest -Uri '$uri' -CustomMethod POST -Body 'testparam=testvalue'"
+    It "Validate Invoke-WebRequest default ContentType for CustomMethod <method>" -TestCases @(
+        @{method = POST}
+        @{method = PUT}
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:757]: Should it be:
```suggestion
        @{method = "POST"}
        @{method = "PUT"}
```

Comment (CarloToso) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:757]: Ops

**Comment by iSazonov** [comment]
@CarloToso Please remove unrelated information from PR description.

**Review by iSazonov — APPROVED** [review]

**Comment by CarloToso** [comment]
@iSazonov can we merge this? Do we need documentation?

**Comment by iSazonov** [comment]
Yes, please open doc issue and update the PR description to make it more clear for doc writer.

**Pull request merged**